internal/core/model/adapters.rs

```diff
@@ -413,3 +415,510 @@ fn test_filter_model() {
+        if count == 0 {
+            return;
+        }
+
+        let mut removed_rows = vec![];
+
+        let mut i = 0;
+
+        loop {
+            if i >= self.mapping.borrow().len() {
+                break;
+            }
+
+            let sort_index = *self.mapping.borrow().get(i).unwrap();
+
+            if sort_index >= index {
+                if sort_index < index + count {
+                    removed_rows.push(i);
+                    self.mapping.borrow_mut().remove(i);
+                    println!("test");
+                    continue;
+                } else {
+                    *self.mapping.borrow_mut().get_mut(i).unwrap() -= count;
+                }
+            }
+
+            i += 1;
+        }
+
+        for removed_row in removed_rows {
+            self.notify.row_removed(removed_row, 1);
+        }
+    }
+
+    fn reset(&self) {
+        self.build_mapping_vec();
+        self.notify.reset();
+    }
+}
+
+/// Provides a sorted subset of rows by another [`Model`].
+///
+/// When the other Model is updated, the `Sorted` is updated accordingly.
+///
+/// ## Example
+///
+/// Here we have a [`VecModel`] holding [`SharedString`]s.
+/// It is then sorted into a `SortModel`.
+///
+/// ```
+/// # use slint::{Model, VecModel, SharedString, SortModel};
+/// let model = VecModel::from(vec![
+///     SharedString::from("Lorem"),
+///     SharedString::from("ipsum"),
+///     SharedString::from("dolor"),
+/// ]);
+///
+/// let sorted_model = SortModel::new(model, |lhs, rhs| lhs.to_lowercase().cmp(&rhs.to_lowercase()));
+///
+/// assert_eq!(sorted_model.row_data(0).unwrap(), SharedString::from("dolor"));
+/// assert_eq!(sorted_model.row_data(1).unwrap(), SharedString::from("ipsum"));
+/// assert_eq!(sorted_model.row_data(2).unwrap(), SharedString::from("Lorem"));
+/// ```
+///
+/// Alternatively you can use the shortcut [`ModelExt::sort_by`].
+/// ```
+/// # use slint::{Model, ModelExt, VecModel, SharedString, SortModel};
+/// let sorted_model = VecModel::from(vec![
+///     SharedString::from("Lorem"),
+///     SharedString::from("ipsum"),
+///     SharedString::from("dolor"),
+/// ]).sort_by(|lhs, rhs| lhs.to_lowercase().cmp(&rhs.to_lowercase()));
+/// # assert_eq!(sorted_model.row_data(0).unwrap(), SharedString::from("dolor"));
+/// # assert_eq!(sorted_model.row_data(1).unwrap(), SharedString::from("ipsum"));
+/// # assert_eq!(sorted_model.row_data(2).unwrap(), SharedString::from("Lorem"));
+/// ```
+/// 
+/// It is also possible to get a ascending sorted  `SortModel` order for `std::cmp::Ord` type items.
+///
+/// ```
+/// # use slint::{Model, VecModel, SortModel};
+/// let model = VecModel::from(vec![
+///     5,
+///     1,
+///     3,
+/// ]);
+///
+/// let sorted_model = SortModel::new_ascending(model);
+///
+/// assert_eq!(sorted_model.row_data(0).unwrap(), 1);
+/// assert_eq!(sorted_model.row_data(1).unwrap(), 3);
+/// assert_eq!(sorted_model.row_data(2).unwrap(), 5);
```

Comment:
You hid the `assert_eq` elsewhere. Did you leave them visible here (and in lines 676-684) on purpose?

Reply:
Ok thank you.